### PR TITLE
Preparing for UDM2.1

### PR DESCRIPTION
**Pull request by mikhailklassen**
This PR includes changes across all UDM and UDM2 notebooks to bring them up to speed. The upcoming launch of UDM2.1 will reduce the number of haze classes from two to one. All detected haze will now be stored in the `light_haze` channel, whereas the `heavy_haze` channel will still be present for backwards compatibility but zeroed.

Changes implemented in this PR:
- Consistent authentication with the Planet API via an API Key
- Consistent use of the Planet Python SDK for accessing Planet image assets
- Combined the UDM and UDM2 notebooks under a single folder. This is because the older UDM asset is still supported, but is shipped in the last channel of the UDM2 asset. The UDM asset is still valuable for diagnosing pixel quality issues.
- Bug and performance fixes to the Python code, especially for activating and downloading image assets
- Added documentation around UDM2.1 and included suitable examples

### CODE REVIEW

File: jupyter-notebooks/udm2/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Working with Usable Data Mask (UDM2)
+# Working with the Unusable Data Mask (UDM) and Usable Data Mask (UDM2)
 
-These notebooks demonstrate the basics of visualizing and working with the Usable Data Mask (UDM2). Specifications for the UDM2 can be found in the [Imagery Product Spec Sheet](https://assets.planet.com/docs/combined-imagery-product-spec-april-2019.pdf).
+These notebooks demonstrates the basics of visualizing and working with the Unusable Data Mask (UDM) and Usable Data Mask (UDM2). Specifications for these can be found in the [Imagery Product Spec Sheet](https://assets.planet.com/docs/Planet_Combined_Imagery_Product_Specs_letter_screen.pdf).
```
Comment (cameronbronstein) [inline jupyter-notebooks/udm2/README.md:3]: demonstrate

File: jupyter-notebooks/udm2/udm2.ipynb [hunk 2]
```diff
@@ -6,18 +6,22 @@
...
+    "The UDM2 is a multi-band image. The first six bands are boolean bands that encode the absence or presence of different image usability concerns such as haze, clouds, and cloud shadows. The seventh band provides the confidence in the usability assessments as integer values between 0 and 100. The eighth band is the Unusable Data Mask, which is described in the Imagery Spec Sheet referenced above and explored in the [UDM notebook](./udm.ipynb).\n",
     "\n",
-    "In this notebook, we demonstrate how to visualize the UDM2 bands and how to convert the UDM2 into two masks: masking all unclear pixels and masking cloud and shadow pixels."
+    "In this notebook, we demonstrate how to visualize the UDM2 bands and how to convert the UDM2 into two masks: masking all unclear pixels and masking cloud and shadow pixels.\n",
```
Comment (cameronbronstein) [inline jupyter-notebooks/udm2/udm2.ipynb:13]: change "shadow" to "cloud shadow"

File: jupyter-notebooks/udm2/udm2.ipynb [hunk 3]
```diff
@@ -6,18 +6,22 @@
...
+    "\n",
+    "### Note on UDM2.1\n",
+    "\n",
+    "After November 2023, with the launch of UDM2.1, the heavy haze class has been deprecated. A new computer vision model trained at Planet returns more accurate class masks, and includes only a single haze class. For backwards compatibility, `light_haze` will store this haze detection, whereas `heavy_haze` will return zero for all pixels going forward."
```
Comment (cameronbronstein) [inline jupyter-notebooks/udm2/udm2.ipynb:17]: maybe we can link to the product spec here (I know it's not available yet).

I also think you can be a bit clearer about the fact that scenes captured before November 29, 2023 are UDM2.0 and have a valid heavy haze band, whereas after the release date, the UDM2.1 masks will all have a zeroed heavy_haze band. 

Maybe something like:

> "An updated Usable Data Mask, UDM2.1, was released on November 29, 2023: all imagery collected after this date will be processed with UDM2.1. For the new product version, the computer vision model that powers UDM2 has been updated to deliver more accurate cloud masks. The UDM2.1 includes only a single haze class, and the heavy haze class has been deprecated.  For backwards compatibility, the `light_haze` band will store the new haze information. Pixels will never be classified as Heavy Haze with UDM2.1, and the `heavy_haze` band will contains zeros for all pixels. All imagery collected before the release date has been processed with UDM2.0 cloud masks, and contains both haze bands."

File: jupyter-notebooks/udm2/udm2_clouds.ipynb [hunk 4]
```diff
@@ -278,13 +278,7 @@
...
    "outputs": [],
    "source": [
-    "# wait until downloads complete\n",
-    "r1.wait()\n",
-    "r2.wait()\n",
-    "img_file = os.path.join(data_dir, r1.get_body().name)\n",
-    "udm_file = os.path.join(data_dir, r2.get_body().name)\n",
-    "print(\"image: {}\".format(img_file))\n",
-    "print(\"udm2:  {}\".format(udm_file))"
+    "udm_file = await download_asset('PSScene', item_id, 'ortho_udm2', data_folder)"
```
Comment (cameronbronstein) [inline jupyter-notebooks/udm2/udm2_clouds.ipynb:281]: maybe udm2_file?

File: jupyter-notebooks/udm2/udm2_clouds_aoi.ipynb [hunk 5]
```diff
@@ -297,10 +258,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# start downloads\n",
-    "data_dir = 'data'\n",
-    "r1 = client.download(assets[\"ortho_analytic_4b\"], callback=api.write_to_file(data_dir))\n",
-    "r2 = client.download(assets[\"ortho_udm2\"], callback=api.write_to_file(data_dir))"
+    "# Activate and download UDM asset\n",
```
Comment (cameronbronstein) [inline jupyter-notebooks/udm2/udm2_clouds_aoi.ipynb:261]: Change UDM to UDM2, udm_file to udm2_file

**Review by cameronbronstein — APPROVED** [review]
Left a few comments but in general this looks goood.

**Review by mikhailklassen — COMMENTED** [review]
Reviewed all comments and made relevant changes.

**Pull request merged**